src/bucket/Bucket.cpp

```diff
@@ -77,16 +77,33 @@ Bucket::Bucket()
 {
 }
 
+void
+Bucket::openStream(std::unique_ptr<XDRInputFileStream>& streamPtr)
+{
+    releaseAssertOrThrow(!streamPtr);
```

Comment:
nit: just return a unique_ptr  and you won't need to worry about verifying the reference. It's generally a bit weird style.